File: walletsdk/src/main/java/leaf/prod/walletsdk/model/TrendInterval.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2018-09-19 下午2:32
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model;

import com.google.gson.annotations.SerializedName;

public enum TrendInterval {

    @SerializedName("1Hr")
    ONE_HOUR("1Hr"),

    @SerializedName("2Hr")
    TWO_HOURS("2Hr"),

    @SerializedName("4Hr")
    FOUR_HOURS("4Hr"),

    @SerializedName("1Day")
    ONE_DAY("1Day"),

    @SerializedName("1Week")
    ONE_WEEK("1Week");

    private String description;

    TrendInterval(String description) {
        this.description = description;
    }

    public static TrendInterval getInterval(String description) {
        TrendInterval result = TrendInterval.ONE_DAY;
        for (TrendInterval interval : TrendInterval.values()) {
            if (interval.getDescription().equalsIgnoreCase(description)) {
                result = interval;
                break;
            }
        }
        return result;
    }

    public String getDescription() {
        return description;
    }
}

File: loopr-ios/Models/Market/TrendInterval.swift
//
//  TrendInterval.swift
//  loopr-ios
//
//  Created by xiaoruby on 12/1/18.
//  Copyright © 2018 Loopring. All rights reserved.
//

import Foundation

enum TrendInterval: String, CustomStringConvertible {
    
    case oneMinutes = "OHLC_INTERVAL_ONE_MINUTES"
    case fiveMinutes = "OHLC_INTERVAL_FIVE_MINUTES"
    case fourHours = "4Hr"
    case oneDay = "1Day"
    case oneWeek = "1Week"
    
    var description: String {
        switch self {
        case .oneHour: return LocalizedString("1Hr", comment: "")
        case .twoHours: return LocalizedString("2Hr", comment: "")
        case .fourHours: return LocalizedString("4Hr", comment: "")
        case .oneDay: return LocalizedString("1Day", comment: "")
        case .oneWeek: return LocalizedString("1Week", comment: "")
        }
    }
    
}
